cpu/esp_common/include/syscalls_common.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2019 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_esp_common
+ * @{
+ *
+ * @file
+ * @brief       Implementation of required system calls for ESP SoCs
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ *
+ * @}
+ */
+
+#ifndef SYSCALLS_COMMON_H
+#define SYSCALLS_COMMON_H
+
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "esp_common.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef DOXYGEN
+/** Alias for source code compatibility with ESP-IDF */
+#define get_free_heap_size esp_get_free_heap_size
+#endif
+
+/** Necessary initializations of system call functions */
+void syscalls_init (void);
+
+/** Determine free heap size */
+unsigned int get_free_heap_size (void);
+
+/** Time since boot in us (32bit version) */
+uint32_t system_get_time (void);
+
+/** Time since boot in ms (32bit version) */
+uint32_t system_get_time_ms (void);
+
+/** memset version that the compiler should not be allowed to optimize this */
```

Comment:
Drop "this"?

Reply:
This system functions are used in vendor code and platform implementation as well. I had to declare them anywhere. So I though to place all system function prototypes in `syscalls.h` which is included in a lot of files. This was the same before the deduplication.

Reply:
I mean to rephrase the comment like below i.e. without "this" at the end:

"memset version that the compiler should not be allowed to optimize"

Reply:
Ah, I see.